### DISCUSSION TITLE
Write JS Arrow tables to node DuckDB

### DISCUSSION THREAD

**Discussion by glacial-ai** [duckdb/duckdb#8680] [Q&A]
Is there a way to write an Arrow table to DuckDB using the node API? For my use-case I'm reading in JSON data to JS Arrow tables (using `arrow.tableFromJSON`) but can't figure out how write them to DuckDB. I've reviewed the node test script for arrow but it's quite limited. Further, there is a string in the test script saying `register buffer should be disabled currently`, so I'm not sure how else I can register an Arrow table and/or record batches. Every time I try to use `db.register_buffer` the script exits abruptly. Here is an example of what I'm trying to do:
```
const arrow = require('apache-arrow');
const duckdb = require('duckdb');

db = new duckdb.Database('path/to/somedb.db');
connection = db.connect();

const table = arrow.tableFromJSON([
    {'a': 1, 'b': 4},
    {'a': 2, 'b': 5},
    {'a': 3, 'b': 6}
]);

// Instead of just passing `table` here, I have also tried:
//    arrow.tableToIPC(table)
//    Buffer.from(arrow.tableToIPC(table))
//    table.toArray()
//    ... nothing seems to work
connection.register_buffer("my_table", table , true, (err) => {
    console.error(err);
});
console.log('buffer registered') // this line never gets printed

connection.all('select * from my_table', (err, result) => {
    assert(!err);
    console.log(result)
})

console.log('success!')
```

Any help would be appreciated!

**Comment by paulrutter** [comment]
@glacial-ai i'm looking into something similar (https://github.com/duckdb/duckdb/discussions/8466), and haven't got it to work on Windows, it does work on Unix.
I tried your code, and it proceeds when making the second argument an array instead.

```
    db.exec(`INSTALL arrow; LOAD arrow;`, (err) => {
      if (err) {
        console.error(err);
      }
      db.register_buffer("my_table", [arrow.tableToIPC(table)], true, (err2, res) => {
        if (err2) {
          console.error(err2);
        }
      });
    });
```

Still, it doesn't work for me on Windows because it cannot load the ARROW extension:
` Error: HTTP Error: Failed to download extension "arrow" at URL "http://extensions.duckdb.org/v0.8.1/windows_amd64/arrow.duckdb_extension.gz"` 
Created https://github.com/duckdb/duckdb/issues/8758 for this.
